### PR TITLE
Document how to test OIDC with DevServices and minor updates to quarkus-test-keycloak-server

### PR DESCRIPTION
**Pull request by sberyozkin**
This PR:
- documents how to test OIDC `service` and `web-app` applications with `Dev Services For Keycloak`
- updates `KeycloakTestResourceLifecycleManager` to disable `Dev Services For Keycloak` as proposed by Stuart
- changes `KeycloakTestClient` to have a `client` (instead of `server`) sub-package

See also: https://github.com/quarkusio/quarkus-quickstarts/pull/928

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]
LGTM

File: docs/src/main/asciidoc/security-openid-connect-dev-services.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect-dev-services.adoc:14]:
```suggestion
Quarkus introduces an experimental `Dev Services For Keycloak` feature which is enabled by default when the `quarkus-oidc` extension is started in dev mode with `mvn quarkus:dev` and when the integration tests are running in test mode, but only when no `quarkus.oidc.auth-server-url` property is configured.
```

File: docs/src/main/asciidoc/security-openid-connect-dev-services.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect-dev-services.adoc:15]:
```suggestion
It starts a Keycloak container for both the dev and/or test modes and initializes them by registering the existing Keycloak realm or creating a new realm with the client and users for you to start developing your Quarkus application secured by Keycloak immediately. It will restart the container when the `application.properties` or the realm file changes have been detected.
```

File: docs/src/main/asciidoc/security-openid-connect-dev-services.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect-dev-services.adoc:49]:
```suggestion
Note that you can disable sharing the containers with `quarkus.keycloak.devservices.shared=false`.
```

File: docs/src/main/asciidoc/security-openid-connect-dev-services.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect-dev-services.adoc:81]:
```suggestion
If Keycloak does enforce it then you will see an authentication error informing you that the `redirect_uri` value is wrong.
```

File: docs/src/main/asciidoc/security-openid-connect-dev-services.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect-dev-services.adoc:83]:
```suggestion
In this case select the `Keycloak Admin` option in the right top corner, login as `admin:admin`, select the test realm and the client which Dev UI for Keycloak is configured with and add `http://localhost:8080/q/dev/io.quarkus.quarkus-oidc/provider` to `Valid Redirect URIs`. If you used `-Dquarkus.http.port` when starting Quarkus then change `8080` to the value of `quarkus.http.port`.
```

File: docs/src/main/asciidoc/security-openid-connect-dev-services.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect-dev-services.adoc:87]:
```suggestion
You can set the `redirect_uri` value to `*` only for test purposes, especially when the containers are shared between multiple applications.
```

File: docs/src/main/asciidoc/security-openid-connect-dev-services.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect-dev-services.adoc:89]: I think you should start by that maybe. Stating that if you just use the default, it's configured like that?

File: docs/src/main/asciidoc/security-openid-connect-dev-services.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect-dev-services.adoc:136]:
```suggestion
You can run the tests against a Keycloak container started in test mode in link:continuous-testing[Continuous Testing] mode.
```
But to be honest, I don't see the point of this sentence.

File: docs/src/main/asciidoc/security-openid-connect-web-authentication.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect-web-authentication.adoc:790]:
```suggestion
Using link:security-openid-connect-dev-services[Dev Services for Keycloak] is recommended for integration testing against Keycloak.
```

File: docs/src/main/asciidoc/security-openid-connect-web-authentication.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect-web-authentication.adoc:793]:
```suggestion
First prepare your `application.properties`. You can start with a completely empty `application.properties` as `Dev Services for Keycloak` will register `quarkus.oidc.auth-server-url` pointing to the running test container as well as `quarkus.oidc.client-id=quarkus-app` and `quarkus.oidc.credentials.secret=secret`.
```

File: docs/src/main/asciidoc/security-openid-connect-web-authentication.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect-web-authentication.adoc:810]:
```suggestion
Finally write your test the same way as it is described in the <<integration-testing-wiremock, Wiremock>> section above.
```

File: docs/src/main/asciidoc/security-openid-connect-web-authentication.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect-web-authentication.adoc:823]:
```suggestion
If you need to do some integration testing against Keycloak then you are encouraged to do it with <<integration-testing-keycloak-devservices,Dev Services For Keycloak>>.
```

File: docs/src/main/asciidoc/security-openid-connect-web-authentication.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect-web-authentication.adoc:839]:
```suggestion
And configure the Maven Surefire plugin as follows:
```

File: docs/src/main/asciidoc/security-openid-connect.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect.adoc:608]:
```suggestion
Using link:security-openid-connect-dev-services[Dev Services for Keycloak] is recommended for integration testing against Keycloak.
```

File: docs/src/main/asciidoc/security-openid-connect.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect.adoc:622]:
```suggestion
which provides a utility class `io.quarkus.test.keycloak.client.KeycloakTestClient` you can use in tests for acquiring access tokens.
```

File: docs/src/main/asciidoc/security-openid-connect.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect.adoc:624]:
```suggestion
Next prepare your `application.properties`. You can start with a completely empty `application.properties` as `Dev Services for Keycloak` will register `quarkus.oidc.auth-server-url` pointing to the running test container as well as `quarkus.oidc.client-id=quarkus-app` and `quarkus.oidc.credentials.secret=secret`.
```

File: docs/src/main/asciidoc/security-openid-connect.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect.adoc:641]:
```suggestion
Finally write your test which will be executed in JVM mode:
```

File: docs/src/main/asciidoc/security-openid-connect.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect.adoc:675]:
```suggestion
and in native mode:
```

File: docs/src/main/asciidoc/security-openid-connect.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect.adoc:693]:
```suggestion
If you need to do some integration testing against Keycloak then you are encouraged to do it with <<integration-testing-keycloak-devservices,Dev Services For Keycloak>>.
```

File: docs/src/main/asciidoc/security-openid-connect.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-openid-connect.adoc:709]:
```suggestion
And configure the Maven Surefire plugin as follows:
```

**Review by gsmet — DISMISSED** [review]

**Comment by sberyozkin** [comment]
@gsmet Thanks for the review

**Comment by sberyozkin** [comment]
@stuartwdouglas I'll wait until you are back online before I merge, in case you'd like to comment on something. We can always improve these docs going forward as well, thanks

**Comment by sberyozkin** [comment]
Hi All, let me merge now - I'll be following up with further updates shortly anyway

**Pull request merged**
